Improve readme file (#22)
vinibispo/ruby.nvim: README.md

Ruby development plug-in for Neovim. Highly unstable.

# Topics

:large_blue_diamond: [Motivation](#motivation)
:large_blue_diamond: [Installation](#installation)
:large_blue_diamond: [Setup](#setup)
:large_blue_diamond: [Commands](#commands)

---

> 

Reviewer: Total detail… but I don't think these horizontal dividers help a lot (same for similar lines). The following heading top margin and bottom border create a nice visual organization already IMHO.

Author: I'll fix it

  requires = {
    'nvim-lua/plenary.nvim',
    'nvim-treesitter/nvim-treesitter'
  }
}
```

Example with [vim-plug](https://github.com/junegunn/vim-plug):

```vim
Plug 'nvim-lua/plenary.nvim'
Plug 'nvim-treesitter/nvim-treesitter'
Plug 'vinibispo/ruby.nvim'
```

---

# Setup

## Default config

Ruby.nvim comes with sane defaults so that you can get going without having to add much to your config

```lua
local ruby_nvim = require("ruby_nvim")

ruby_nvim.setup({
  test_cmd = "ruby" -- command that :RubyTest is gonna run,
  test_args = {} -- args that :RubyTest is gonna run
})
```

---

> # Commands

Reviewer: 100% a matter of style and personal preference, but I think this structure involving three levels of headings (_Commands, Run, :RubyRun_, for example) makes it difficult to read and to get an overall of what the plugin offers.

I tend to prefer a concise table, as in `yaml.nvim`:

> | Command | Description |
> |:--|:--|
> | `:YAMLView` | Shows the full path and value of the current key/value pair |
> | `:YAMLYank [register]` | Yanks the full path and value of the current key/value pair. The default register is the unnamed one (`"`) |
> | `:YAMLYankKey [register]` | Yanks the full path of the key for the current key/value pair. The default register is the unnamed one (`"`) |
> | `:YAMLYankValue [regster]` | Yanks the value of the current key/value pair. The default register is the unnamed one (`"`) |
> | `:YAMLQuickfix` | Generates a quickfix with key/value pairs |
> | `:YAMLTelescope` | Full path key/value fuzzy finder via [Telescope](https://github.com/nvim-telescope/telescope.nvim) **if installed** |

Author: I'll fix it

## Run

### :RubyRun

> Runs given file (current or another) in a floating window

Reviewer: Why is this different from [the Vim doc](https://github.com/vinibispo/ruby.nvim/blob/debfa50ac82bff053c803f030c40f064e26f2a60/doc/help.txt#L54)?

Author: Because we have written wrong in vimdoc

Reviewer: Got it! Since we expect an optional argument, I think we can have `:RubyRun [file]` instead of just `:RubyRun` ; )

Author: Ok, I'mma change it

  requires = {
    'nvim-lua/plenary.nvim',
    'nvim-treesitter/nvim-treesitter'
  }
}
```

Example with [vim-plug](https://github.com/junegunn/vim-plug):

```vim
Plug 'nvim-lua/plenary.nvim'
Plug 'nvim-treesitter/nvim-treesitter'
Plug 'vinibispo/ruby.nvim'
```

# Setup

## Default config

Ruby.nvim comes with sane defaults so that you can get going without having to add much to your config

```lua
local ruby_nvim = require("ruby_nvim")

ruby_nvim.setup({
  test_cmd = "ruby" -- command that :RubyTest is gonna run,
  test_args = {} -- args that :RubyTest is gonna run
})
```

# Commands

| Command | Description |
|:--|:--|

> | `:RubyRun [file]` | Runs given file (current or another) in a floating window |

Reviewer: Wording suggestion:

```suggestion
| `:RubyRun [file]` | Runs `file` in a floating window (default: current file). |
```

Author: I'mma fix it

> | `:RubyAlternate` | Alternate between a Ruby file and its test file. |

Reviewer: Wording suggestion:

```suggestion
| `:RubyAlternate` | Alternate between a Ruby file and its test file, and vice-versa. |
```

Author: I'mma fix it

| `:RubyTest[!]` | Runs the test for the current file. If the file is not a test file, it tries to to find the (alternate) test file. If run with `!` and it is a test files, runs only the test under the cursor. |

> | `:RubyBrowseGem` | Searches gem that is on cursor line in https://rubygems.org |

Reviewer: Wording suggestion:

```suggestion
| `:RubyBrowseGem` | Searches gem under the cursor line in [rubygems.org](https://rubygems.org). |
```
